### DISCUSSION TITLE
Migration from Capacitor 2.x to 3.x Error iOS: Thread 1: "-[__NSCFNumber isEqualToString:]: unrecognized selector sent to instance 0x93d35cc60a1b4601"

### DISCUSSION THREAD

**Discussion by eduboxgithub** [ionic-team/capacitor#5182] [General]
I'm only now migrating my apps to Capacitor 3.x.

I followed all the steps and tested the app on Android and everything seems to be ok.
Now i'm testing the app on iOS, but I'm getting a strange error.

The app builds from Xcode, is deployed to the device but when the app opens it crash on the splash screen.

In the Xcode I get the following error:

```
Thread 1: "-[__NSCFNumber isEqualToString:]: unrecognized selector sent to instance 0x93d35cc60a1b4601"
```
![image](https://user-images.githubusercontent.com/13800439/139228781-5d7155c4-70ad-4f9c-8704-dc556b977e53.png)

I'm not a native iOS developer, so I don't know exactly what could be wrong.
Any hint on what could be the problem?

I'm using macOS Catalina 10.15.7, xCode 12.2.

```
💊   Capacitor Doctor  💊 

Latest Dependencies:
  @capacitor/cli: 3.2.5
  @capacitor/core: 3.2.5
  @capacitor/android: 3.2.5
  @capacitor/ios: 3.2.5

Installed Dependencies:
  @capacitor/cli: 3.2.5
  @capacitor/core: 3.2.5
  @capacitor/android: 3.2.5
  @capacitor/ios: 3.2.5
  
  [success] iOS looking great! 👌
```

**Comment by jcesarmobile** [comment]
hard to tell with that little information, but I guess some plugin or custom code is getting a number where it expects a string and isEqualToString doesn't exist on numbers.

**Reply by eduboxgithub** [reply]
Thank you @jcesarmobile .

I know... and I'm sorry :( it's not easy to "debug" when no code was provided, but it's hard to share and replicable and example of a migration processe... I was just asking for a hint and you gave me one.

I will review every upgrade and plugin integration step and let you know if I could solved it.

Thank you for your time.
